torch/_dynamo/eval_frame.py

```diff
@@ -789,6 +793,75 @@ def run_node(self, n):
                 r.node.meta["val"] = self.current_node.meta["val"]
             return r
 
+    class AddRuntimeAssertsInInputConstraint(torch.fx.interpreter.Transformer):
```

Comment:
This seems complicated and I am not entirely sure what it does and I don't see it described in PR desc

Reply:
Just updated the PR description. It basically convert input constraints into runtime asserts per specified dimension.